### PR TITLE
Updated the PUT endpoint

### PR DESCRIPTION
**Pull request by ttingle-ch**
__Short description outlining key changes/additions__
- Updated the PUT endpoint to return 500 status code if an error occurs.
- Updated PUT endpoint to stop it updating the transaction api if the Company name/number/business name does not change.
__JIRA Ticket Number__
IDVA5-1347
### Type of change

* [ ] Bug fix
* [ ] New feature
* [ ] Breaking change
* [ ] Infrastructure change

### Pull request checklist

* [ ] I have added unit tests for new code that I have added
* [ ] I have added/updated functional tests where appropriate
* [ ] The code follows our [coding standards](https://github.com/companieshouse/styleguides/blob/master/java.md)

__An exhaustive list of peer review checks can be read [here](https://github.com/companieshouse/styleguides/blob/master/java_review.md#developer-actions-prior-to-code-commitreview-started)__

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/main/java/uk/gov/companieshouse/acsp/service/AcspService.java [hunk 1]
```diff
@@ -103,12 +103,17 @@ public ResponseEntity<Object> updateACSPDetails(Transaction transaction,
...
 
         if (acspDataDto.getCompanyDetails() != null) {
-            transaction.setCompanyName(acspDataDto.getCompanyDetails().getCompanyName());
-            transaction.setCompanyNumber(acspDataDto.getCompanyDetails().getCompanyNumber());
-            transactionService.updateTransaction(requestId, transaction);
+            if(!acspDataDto.getCompanyDetails().getCompanyName().equals(transaction.getCompanyName()) ||
+                    !acspDataDto.getCompanyDetails().getCompanyNumber().equals(transaction.getCompanyNumber())) {
+                transaction.setCompanyName(acspDataDto.getCompanyDetails().getCompanyName());
+                transaction.setCompanyNumber(acspDataDto.getCompanyDetails().getCompanyNumber());
+                transactionService.updateTransaction(requestId, transaction);
```
Comment (sampankumar) [inline src/main/java/uk/gov/companieshouse/acsp/service/AcspService.java:110]: This is common operation I believe so it can come out of if else

**Review by sampankumar — DISMISSED** [review]

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by sampankumar — APPROVED** [review]

**Review by swalke-ch — APPROVED** [review]

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Pull request merged**